ScreenRecorderLib/internal_recorder.cpp

```diff
@@ -1242,6 +1252,20 @@ HRESULT internal_recorder::InitializeVideoSinkWriter(std::wstring path, _In_opt_
 	UINT destWidth = max(0, destRect.right - destRect.left);
 	UINT destHeight = max(0, destRect.bottom - destRect.top);
 
+	UINT outputWidth = destWidth, outputHeight = destHeight;
+	if (m_MaxVideoHeight != 0 || m_MaxVideoWidth != 0) {
+		double aspect = (double)destWidth / (double)destHeight;
+
+		if (m_MaxVideoWidth != 0 && destWidth > m_MaxVideoWidth) {
+			outputHeight = m_MaxVideoWidth / aspect;
+			outputWidth = m_MaxVideoWidth;
+		}
+		else if(m_MaxVideoHeight != 0 && destHeight > m_MaxVideoHeight){
```

Comment:
I think this should be another `if` (not `else if`), otherwise there are some combinations of inputs that could result in output dimensions greater than the specified maximums. For example:
```
// Inputs:
m_MaxVideoWidth = 200
m_MaxVideoHeight = 100

destWidth = 100
destHeight = 1000
aspect = 0.1

// After the first if block
outputHeight = 100 / 0.1 = 1000
outputWidth = 200

// If the second block was another if block, it would run since destHeight > m_MaxVideoHeight
outputWidth = 100 * 0.1 = 10
outputHeight = 100
```

Reply:
Nice catch. I stupidly merged the PR before, so I'll file another one with this.